server/services/offer.service.js

```diff
@@ -308,6 +308,7 @@ export const generatePaymentSchedules = (offer) => {
 };
 
 export const createOffer = async (offer) => {
+  let status = OFFER_STATUS.GENERATED;
```

Comment:
This can be generated in a single liner (I have mentioned this severally).

```suggestion
const status = offer.additionalClause ? OFFER_STATUS.PENDING_APPROVAL : OFFER_STATUS.GENERATED;
```